### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by NATASHA-ct**
### We implemented the following project requirements ✔️ 

- [x] Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
- [x] On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. `rentals.json`)
- [x] On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: app/dashboard.rb [hunk 1]
```diff
@@ -114,11 +74,86 @@ def options(input)
     end
```
Comment (julie-ify) [inline app/dashboard.rb:78]: - [x] But for the data storage implementation, your app is still expected to work like the previous basic UI project. I am unable to exit the app. Kindly implement option 7 which allows users to quit the program peacefully.

File: app/dashboard.rb [hunk 2]
```diff
@@ -114,11 +74,86 @@ def options(input)
...
+      @my_app.people.each { |person| people_json << JSON.generate(person.export_json) }
+      rentals_json = []
+      @my_app.rentals.each { |rental| rentals_json << JSON.generate(rental.export_json) }
+
+      # write data to their respective files
+      File.write('books.json', books_json)
+      File.write('people.json', people_json)
+      File.write('rentals.json', rentals_json)
+    end
+  end
```
Comment (julie-ify) [inline app/dashboard.rb:104]: - [ ] You should handle scenarios for when the files do not exist. When I remove all storage files and start your app. After all data creation when I quit the app the files should be created and the data should be stored in them but that is not the case now. Kindly implement that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @NATASHA-ct and @fahimy143,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/dashboard.rb (hunk 2 above)
Comment (fahimy143) [reply app/dashboard.rb:104]: @julie-ify Kindly explain a little more about this which file do you mean??

File: app/dashboard.rb (hunk 1 above)
Comment (rloterh) [reply app/dashboard.rb:78]: well resolved! ✔️

File: app/dashboard.rb (hunk 2 above)
Comment (rloterh) [reply app/dashboard.rb:104]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Tascha & Fahimy,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Data is stored correctly in JSON on program exit
- JSON files generated on exit of program

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**